examples/fairmotion_interface_utils.py

```diff
@@ -0,0 +1,265 @@
+                    j = i
+
+                # fill root_orientations
+                for pose in motion_.poses:
+                    root_T = pose.get_transform(ROOT, local=False)
+                    root_rotation = mn.Quaternion.from_matrix(
+                        mn.Matrix3x3(root_T[0:3, 0:3])
+                    )
+                    self.root_orientations.append(root_rotation)
+
+                # get center of drift
+                summ = mn.Vector3()
+                for pose in motion_.poses:
+                    root_T = mn.Matrix4(pose.get_transform(ROOT, local=False))
+                    root_T.translation *= (
+                        mn.Vector3(1.0, 1.0, 1.0) - self.direction_forward
+                    )
+                    summ += root_T.translation
+                self.center_of_root_drift = summ / self.num_of_frames
+
+            if type_ == MType.SCENIC:
+                # primary
+                self.motion = motion_
+                self.type = MType.TRANSITIVE
+                self.poses = motion_.poses
+                self.fps = motion_.fps
+                self.num_of_frames: int = len(motion_.poses)
+                self.time_length = self.num_of_frames * (1.0 / motion_.fps)
+
+    ## TRANSITIVE ##
+    motion_ = amass.load(
+        file="data/fairmotion/amass_test_data/CMU/CMU/10/10_04_poses.npz",
+        bm_path="data/fairmotion/amass_test_data/smplh/male/model.npz",
+    )
+    # all motions must have same fps for this implementation, so use first motion to set global
+    fps = motion_.fps
+
+    """
+    [TESTING]
+    start = 125
+    offset = 0 + 1 - 3 + 6 + 2 + 2 + 2 + 1 + 1 + 1
+    length = 145
+    walk_to_walk = MotionData(motion_ops.cut(motion_, start + offset, start + length))
+    """
+
+    # Walk-to-walk cycle
+    walk_to_walk = MotionData(motion_ops.cut(motion_, 300, 430), MType.TRANSITIVE)
+
+    # Standing pose that must converted (amass -> habitat joint positions)
+    standing_pose = motion_.poses[0]
+
+    ## SCENIC ##
+    motion_ = amass.load(
+        file="data/fairmotion/amass_test_data/CMU/CMU/13/13_08_poses.npz",
+        bm_path="data/fairmotion/amass_test_data/smplh/male/model.npz",
+    )
+    drink_beverage = MotionData(motion_, MType.SCENIC)
+
+
+class PathData:
+    """
+    The PathData class is purposed to instantiate with given path points, and used
+    to manage the path data fro the current path following sequence.
+    """
+
+    def __init__(self, path_points: List[mn.Vector3]) -> None:
+        self.points = path_points
+        self.length = self.calc_path_length(path_points)
+        self.time = 0.0
+
+    # [REFACTOR] I would like to change this to a static method, once I figure out how
+    def calc_path_length(self, path_points: List[mn.Vector3]) -> float:
+        # get path length
+        i, j, summ = 0, 0, 0.0
+        while i < len(path_points):
+            summ += (mn.Vector3(path_points[i] - path_points[j])).length()
```

Comment:
```suggestion
            summ += (path_points[i] - path_points[j]).length()
```
if patu_points really is a list of vectors than the extra call is an unnecessary copy.

Reply:
Decided to remove the typing because this function should be able to take a list of path_points whether it be made up of np arrays or mn vector3s